[CONTENT REVIEW] 16.6 - Alterar a redação dos requisitos do projeto (#11)
betrybe/sd-0x-block17-project-react-redux-starwars-database-filters: README.md

A tabela deve ser renderizada por um componente chamado `<Table />`. Os dados recebidos da API devem ser salvos num campo chamado `data` do `store` e é daí que a tabela deve lê-los. A requisição deve ser feita num componente separado do componente da tabela.

> A tabela deve ter uma primeira linha com os headers e as demais com as informações de cada campo. Essa linha será  considerada nos testes, então caso um teste busque por oito linhas na tabela, serão sete linhas com as informações sobre os planetas e uma referente aos headers.

Reviewer: ```suggestion
A tabela deve ter uma primeira linha com os headers e as demais com as informações de cada campo. Essa linha será considerada nos testes, então caso um teste busque por oito linhas na tabela, serão sete linhas com as informações sobre os planetas e uma referente aos headers.
```

  - A seleção `population | maior que | 100000` - Seleciona somente planetas com mais de 100000 habitantes.
  - A seleção `diameter | menor que | 8000` - Seleciona somente planetas com diâmetro menor que 8000.

> Você deve usar **Redux** para fazer o gerenciamento do estado da aplicação. No `store`, esses valores devem ser salvos nos campos `filters { filterByName: { name }, filterByNumericValues: [{ column, comparison, value }] }`. Para que não haja problemas com o avaliador o estado inicial da chave `filterByNumericValues` deve ser um array vazio, `[]`. Veja um exemplo de como o estado deve ficar quando uma filtragem for feita:

Reviewer: ```suggestion
Você deve usar **Redux** para fazer o gerenciamento do estado da aplicação. No `store`, esses valores devem ser salvos nos campos `filters { filterByName: { name }, filterByNumericValues: [{ column, comparison, value }] }`. Tenha em mente que a avaliação automatizada exige que o estado inicial da chave `filterByNumericValues` seja um array vazio, `[]`. Veja um exemplo de como o estado deve ficar quando uma filtragem for feita:
```

### 4 - Sua página deverá ser carregada com somente um filtro de valores numéricos

> Após clicar no botão com o data-testid `button-filter`, o filtro de valores númericos deve ser atualizado, sendo que no dropdown de colunas não deve mais haver a coluna que já havia sido selecionada. Caso todas as colunas já tenham sido incluidas em filtros anteriores, ao atualizar o filtro nenhuma das colunas devem estar disponíveis. Você deve usar **Redux** para fazer o gerenciamento do estado da aplicação.

Reviewer: ```suggestion
Após clicar no botão com o data-testid `button-filter`, sua aplicação deve permitir que quem a usa adicione um novo filtro numérico sem que precise deletar o filtro anterior. Ou seja: dado que você usa três elementos para configurar o seu filtro numérico, após clicar no botão esses três elementos devem permanecer na tela e funcionais na filtragem da tabela, mas três novos elementos devem ser adicionados para que, assim, quem usa pode selecionar um novo filtro numérico. Além disso, no dropdown de colunas desse novo filtro não deve mais haver a coluna que já foi selecionada no filtro anterior. Caso todas as colunas já tenham sido incluídas em filtros anteriores, ao atualizar o filtro nenhuma das colunas devem estar disponíveis. Você deve usar **Redux** para fazer o gerenciamento do estado da aplicação.
```

O que acham de colocar,  aqui, um gif específico para ilustrar o comportamento esperado da adição de filtros? Acho que pode ajudar bem!

Author: O @KingUrek adicionou, você consegue visualizar como ficou nesse [PR](https://github.com/betrybe/sd-0x-block17-project-react-redux-starwars-database-filters/pull/10).

> Por exemplo: O primeiro filtro tem as seguintes seleções: `population | maior que | 100000`. O filtro deve atualizado após o botão com o data-testid `button-filter` ser clicado e, no primeiro dropdown deste filtro, a opção `population` deve estar ausente. Se no segundo filtro fosse selecionado `diameter | menor que | 8000`, o estado ficaria assim:

Reviewer: ```suggestion
Por exemplo: imagine que você selecionou no primeiro filtro numérico as seguintes seleções: `population | maior que | 100000`. Após clicar no botão com data-testid `button-filter` deve ser possível adicionar um segundo filtro, além deste. No dropdown desse novo filtro a se adicionar, a opção `population` não deve aparecer. Em resumo, seus filtros numéricos não podem repetir as colunas que selecionam. Se no segundo filtro fosse selecionado `diameter | menor que | 8000`, o estado ficaria assim:
```

Que tal assim? Entendo que a descrição original estava bem confusa, mas acho que essa ainda não estava tão clara. Não estou 100% confiante quanto à minha sugestão também. O que acha? 🤔

Author: Gostei como escreveu, o exemplo ficou mais claro. Acredito que dessa forma a pessoa que ler terá mais facilidade de entender como o filtro deve funcionar.

---

### Implementações técnicas

> Algumas coisas devem seguir um padrão pré-estabelecido para que os teste de correção funcionem corretamente.

Reviewer: ```suggestion
Algumas coisas devem seguir um padrão pré-estabelecido para que a avaliação automatizada aprove a implementação.
```

#### Provider

> É preciso que o `store` seja provido para a aplicação no `index.js`. Caso contrário não será possível realizar os testes, pois como você viu para testar uma aplicação que utiliza **Redux** é necessário prover a aplicação um no store, esse que seja possível manipular.

Reviewer: ```suggestion
É preciso que o `store` seja provido para a aplicação no `index.js`. Caso contrário não será possível realizar os testes, pois como você viu para testar uma aplicação que utiliza **Redux** é necessário que os testes provejam à aplicação um store próprio, substituindo um que deve ser declarado no `index.js`.
```

#### Reducers

> Nos testes, para a criação do `store`, é utilizado o `reducer` que você desenvolveu, então para que os testes ocorram como o previsto é preciso que seu `reducer` final seja exportado de `src/reducers/index.js`, pois é desse caminho que o arquivo de testes estará importando o seu `reducer`.

Reviewer: ```suggestion
Nos testes, para a criação do `store`, é utilizado o `reducer` que você desenvolveu, então para que os testes ocorram como o previsto é preciso que seu `reducer` final seja exportado de `src/reducers/index.js`, pois é desse caminho que o arquivo de testes fará a importação dele.
```
